emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/preader/ScanRecordReadRequest.java

```diff
@@ -54,4 +63,44 @@ protected PageResults<Map<String, AttributeValue>> fetchPage(RequestLimit lim) {
         consumedCapacityUnits,
         retries);
   }
+
+  /**
+   * Set a Scan query filter to skip expired records if the configuration
+   * provides the TTL attribute name
+   */
+  Optional<DynamoDBQueryFilter> queryFilter() {
+    Optional<String> maybeTtlAttribute =
+        Optional.ofNullable(context.getConf().get(DynamoDBConstants.TTL_ATTRIBUTE_NAME));
+    return maybeTtlAttribute.map(attributeName -> {
+      long now = Instant.now().getEpochSecond();
```

Comment:
How often is this called?

Reply:
I submitted #11 to evaluate early the Scan Filter, in case that helps.